src/Stratis.Bitcoin.Features.Wallet/WalletManager.cs

```diff
@@ -99,6 +102,7 @@ public class WalletManager : IWalletManager
         internal ScriptToAddressLookup scriptToAddressLookup;
 
         public WalletManager(
+            IBlockStore blockStore,
```

Comment:
I am not sure this is correct, putting store as a dependency in wallet.

Reply:
Access is required to the store to load the transaction from disk. We need to do this to get the various inputs and outputs to determine the address groupings. Not sure how we can get around this?

Reply:
Off the top of my head, Core always stores full info for its own wallet transactions even if txindex is disabled. We may have to adopt a similar methodology, i.e. re-enabling the default saving of transaction hex, perhaps.

Reply:
Doint we already have this data? we do store additional info for a UTXO (I think its called SpendingDetails) @bokobza ?

Reply:
We only store the outputs with `TransactionData`. We could have used the transaction hex but then the wallet settings' `SaveTransactionHex` must be turned on. So this is a bit of catch 22 situation.

We could remove the blockstore dependency and then just check that the wallet's `SaveTransactionHex` setting is true? If not throw an exception?

@zeptin @dangershony @mikedennis WDYT?

Reply:
This is done only in order to get the `prevTransactionScriptPubkey` right? do we not store it in spending details?

Reply:
Yeah I think we do have that `destinationScriptPubKey`